lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -5,13 +5,27 @@ const _ = require('lodash');
 
 module.exports = {
   compileMethods() {
+    const corsConfig = {};
```

Comment:
@pmuens What I'm using that for is to build up the CORS requirements for each http path in the service. As Serverless is function-oriented and CORS is http method-oriented, I needed to build up an object with the appropriate configurations for each path (rather than function). As such, the if-statement lower down is rightly checking to see if there are any existing settings for this http method and to append to them if they do exist, otherwise create a new property on the object corresponding to that path. The keys of this object are finally iterated over after all the functions have been processed and the preflight request (OPTIONS) cf template is set. Does that make sense? There could be a better way of doing this...